subsys/net/lib/lwm2m/lwm2m_util.c

```diff
@@ -528,3 +528,51 @@ bool lwm2m_obj_path_equal(const struct lwm2m_obj_path *a, const struct lwm2m_obj
 
 	return true;
 }
+
+/* for debugging: to print IP addresses */
+char *lwm2m_sprint_ip_addr(const struct sockaddr *addr)
```

Comment:
We have similar functions in core net although they are now "private" and not in public api. I just wonder if we should create for example `<net/utils.h>` that could contain various helper functions so that these would not be needed to implement in the network libraries and/or applications. This is just brainstorming and not part of this PR.